Adds support for NA RRM grid on ELM

bishtgautam opened the pull request:
- Adds northamericax4v1pg2 as a valid resolution
- Adds mapping files needed to create surface datasets at northamericax4v1pg2
- Adds surface dataset for 1850 and 2010

[BFB]
Fixes #4029

jonbob reviewed (approved):
Looks good from a visual inspection. It will get tested by the RRM project once it's in. Thanks for the quick turnaround

golaz commented:
@bishtgautam: thanks for generating these files.

bishtgautam commented:
@jonbob I successfully ran two tests using the newly created surface datasets, but I had to use ELM's user-defined resolution because `-res northamericax4v1_northamericax4v1` isn't supported currently.

jonbob commented:
@bishtgautam - that's great. We'll have to make a PR adding the non-trigrid resolution of the fully-active configuration and should put that one in with it. Thanks again

tangq commented:
Hi @bishtgautam , I am putting together the coupled NA RRM configurations and notice that the land grid name in this PR doesn't match the existing RRM grid naming convention used in the tri-grid configuration.

Can you rename the land grid from northamericax4v1pg2 to ne0np4_northamericax4v1.pg2? Thanks.

bishtgautam commented:
@tangq, I have renamed the files on the input data server and updated them this PR. But, I have not changed `hgrid="northamericax4v1pg2" ` to `hgrid="ne0np4_northamericax4v1.pg2"`. Would you like me to change `hgrid` too?

tangq commented:
Thanks for doing this, @bishtgautam . Yes, we need to change the hgrid names.

bishtgautam commented:
@tangq I have now updated hgrid name.

tangq commented:
@bishtgautam , my 3-day test on the bi-grid RRM was just completed successfully. I will make some in-line comments to this PR regarding the grid name change (It will be more efficient and clearer) to make it compatible with the tri-grid names.

tangq commented on components/elm/bld/namelist_files/namelist_definition.xml:
```diff
@@ -1256,7 +1256,7 @@ CLM run type.
        group="default_settings"  
        valid_values=
-"512x1024,360x720cru,128x256,64x128,48x96,32x64,8x16,94x192,0.23x0.31,0.9x1.25,1.9x2.5,2.5x3.33,4x5,10x15,5x5_amazon,1x1_tropicAtl,1x1_camdenNJ,1x1_vancouverCAN,1x1_mexicocityMEX,1x1_asphaltjungleNJ,1x1_brazil,1x1_urbanc_alpha,1x1_numaIA,1x1_smallvilleIA,0.1x0.1,0.5x0.5,3x3min,5x5min,10x10min,0.33x0.33,ne4np4,ne4np4.pg2,ne11np4,ne16np4,ne30np4,ne30np4.pg2,ne60np4,ne120np4,ne240np4,1km-merge-10min ...
+"512x1024,360x720cru,128x256,64x128,48x96,32x64,8x16,94x192,0.23x0.31,0.9x1.25,1.9x2.5,2.5x3.33,4x5,10x15,5x5_amazon,1x1_tropicAtl,1x1_camdenNJ,1x1_vancouverCAN,1x1_mexicocityMEX,1x1_asphaltjungleNJ,1x1_brazil,1x1_urbanc_alpha,1x1_numaIA,1x1_smallvilleIA,0.1x0.1,0.5x0.5,3x3min,5x5min,10x10min,0.33x0.33,ne4np4,ne4np4.pg2,ne11np4,ne16np4,ne30np4,ne30np4.pg2,ne60np4,ne120np4,ne240np4,1km-merge-10min ...
```
Need to rename northamericax4v1.pg2 to ne0np4_northamericax4v1.pg2 to work with the existing tri-grid configuration.

tangq commented on components/elm/bld/namelist_files/namelist_defaults.xml:
```diff
@@ -1749,6 +1756,39 @@ this mask will have smb calculated over the entire global land surface
 <map frm_hgrid="0.5x0.5"    frm_lmask="GSDTG2000"  to_hgrid="ne240np4"   to_lmask="nomask"
 >lnd/clm2/mappingdata/maps/ne240np4/map_0.5x0.5_GSDTG2000_to_ne240np4_nomask_aave_da_c170821.nc</map>
 
+<map frm_hgrid="0.5x0.5"    frm_lmask="AVHRR"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_0.5x0.5_AVHRR_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="0.5x0.5"    frm_lmask="MODIS"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_0.5x0.5_MODIS_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="0.5x0.5"    frm_lmask="nomask"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_0.5x0.5_nomask_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="10x10min"    frm_lmask="nomask"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_10x10min_nomask_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="3x3min"    frm_lmask="MODIS"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_3x3min_MODIS_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="3x3min"    frm_lmask="USGS"   to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_3x3min_USGS_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="3x3min"    frm_lmask="LandScan2004" to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_3x3min_LandScan2004_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="5x5min"    frm_lmask="IGBP-GSDP"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_5x5min_IGBP-GSDP_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="5x5min"    frm_lmask="ISRIC-WISE"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_5x5min_ISRIC-WISE_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="5x5min"    frm_lmask="nomask"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_5x5min_nomask_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="3x3min"   frm_lmask="GLOBE-Gardner" to_hgrid="northamericax4v1pg2" to_lmask="nomask"
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_3x3min_GLOBE-Gardner_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="3x3min"   frm_lmask="GLOBE-Gardner-mergeGIS" to_hgrid="northamericax4v1pg2" to_lmask="nomask"
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_3x3min_GLOBE-Gardner-mergeGIS_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="0.9x1.25"    frm_lmask="GRDC"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_0.9x1.25_GRDC_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="360x720cru"    frm_lmask="cruncep"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_360x720cru_cruncep_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="1km-merge-10min"    frm_lmask="HYDRO1K-merge-nomask"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_1km-merge-10min_HYDRO1K-merge-nomask_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
+<map frm_hgrid="0.5x0.5"    frm_lmask="GSDTG2000"  to_hgrid="northamericax4v1pg2"   to_lmask="nomask" 
+>lnd/clm2/mappingdata/maps/northamericax4v1pg2/map_0.5x0.5_GSDTG2000_to_northamericax4v1pg2_nomask_aave_da_c210112.nc</map>
```
@bishtgautam , I am not sure if the `to_hgrid` field needs to be renamed or not. It seems to me this session only affects the maps in the preprocessing step, instead of the model simulation. I didn't change it in my bi-grid RRM test, which ran successfully. Change it or leave it as you see fit.

bishtgautam commented:
@tangq I made more fixes to this PR

tangq commented:
It all looks great and ready to be merged.
